realkd/rules.py

```diff
@@ -412,14 +420,14 @@ def bound(self, ext):
     def opt_weight(self, q):
         # TODO: this should probably just be defined for ext (saving the q evaluation)
         # ext = self.ext(q)
-        ext = self.data.loc[q].index
+        ext = q(self.data).nonzero()[0]
```

Comment:
Since this is a common pattern and in my opinion quite hard to read - I think it deserves a function - but I was trying to be as minimal as possible so didn't add anything.

Reply:
I think it is actually quite readable. The only unexpected part is the final `[0]` , which I assume is necessary because numpy returns a matrix instead of a vector?

BTW: The todo comment points to a straightforward performance improvement for this specific function